tests/test_utils/test_simpletestcase.py

```diff
@@ -0,0 +1,84 @@
+import unittest
+from io import StringIO
+
+from django.test import SimpleTestCase
+
+
+class ErrorTestCase(SimpleTestCase):
+    def __init__(self, *args, **kwargs):
+        super(ErrorTestCase, self).__init__(*args, **kwargs)
+        self._running_test = []
+
+    def _pre_setup(self):
+        self._running_test.append(repr(self))
+
+    def _post_teardown(self):
+        self._running_test.pop()
+
+    def raising_test(self):
+        self.assertTrue(self._running_test, '_pre_setup expected to be called')
+        raise Exception('Should bubble up when using debug()')
+
+    def simple_debug_test(self):
+        self.assertTrue(self._running_test, '_pre_setup expected to be called')
+
+    @unittest.skip('Testing debug skip conditions')
+    def skipped_test(self):
+        pass
+
+
+class DebugInvocationTests(SimpleTestCase):
+
+    def get_runner(self):
+        return unittest.TextTestRunner(stream=StringIO())
+
+    def test_post_teardown_called(self):
+        """Simple test run: catches errors and runs cleanup."""
+        test_suite = unittest.TestSuite()
+        case = ErrorTestCase('raising_test')
+        test_suite.addTest(case)
+        test_suite.run(self.get_runner()._makeResult())
+        # _post_teardown removes the object added in _pre_setup()
+        self.assertEqual(case._running_test, [])
```

Comment:
Did you consider using mocking and assertions such as `assert_called_once_with()`? This might be a bit cleaner and more self-explanatory than introspecting `_running_test`.